### PR TITLE
Documentation for API usage

### PR DESCRIPTION
**Pull request by nqb**
# Description
* Add section related to `pfperl-api`  usage
* Improve `curl` examples

# Impacts
* No impacts

# Delete branch after merge
YES

# Checklist
- [x] Document the feature

## Enhancements
* Documentation to set up dev environments

### CODE REVIEW

**Comment by nqb** [comment]
NOT READY

**Comment by nqb** [comment]
See also be5cd36

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 1]
```diff
@@ -1011,23 +1011,85 @@ http://packetfence.org/bugs/view.php?id=1085[issue #1085].
...
+yum groupinstall "Development Tools" -y
+----
+. Run following commands:
++
+[source,bash]
+----
+mv /usr/local/pf/ /usr/local/pf-pkg
+cd /usr/local/
+git clone https://github.com/inverse-inc/packetfence.git pf
+mv conf conf-git-pristine
```
Comment (extrafu) [inline docs/PacketFence_Developers_Guide.asciidoc]: you miss a "cd pf" after the clone

**Comment by nqb** [comment]
When #4663 will be merge, I will review this PR and rebase to keep only things related to API calls. Documentation for devel environment has been integrated into #4663.

File: docs/PacketFence_Developers_Guide.asciidoc [hunk 2]
```diff
@@ -1011,23 +1045,157 @@ http://packetfence.org/bugs/view.php?id=1085[issue #1085].
...
+make conf/ssl/server.pem
+mkdir /usr/local/pf/var/ssl_mutex
+cp ../pf-pkg/conf/pf.conf conf/
+cp ../pf-pkg/conf/pfconfig.conf conf/
+cp ../pf-pkg/conf/currently-at conf/
+ln -s /usr/local/pf/raddb/sites-available/status /usr/local/pf/raddb/sites-enabled/status
+----
++
+
+. Install (at least) link:https://github.com/nodesource/distributions/#table-of-contents[NodeJS] 8 to build JavaScript files for webadmin
```
Comment (cgx) [inline docs/PacketFence_Developers_Guide.asciidoc]: Should be version 10 or even 12.

**Pull request merged**